src/util/TickUtils.ts

```diff
@@ -11,8 +11,55 @@ export interface CartesianTickItem extends TickItem {
   isShow?: boolean;
 }
 
+/**
+ * Given an array and a number N, return a new array which contains every nTh
+ * element of the input array.
+ * @param {any[]} array An input array.
+ * @param {integer} n A number
+ * @returns {any[]} The result array of the same type as the input array.
+ */
+function getEveryNth<Type>(array: Type[], n: number): Type[] {
+  if (n === 1) {
+    return array;
+  }
+  const result = [];
+  for (let i = 0; i < array.length; i += n) {
+    result.push(array[i]);
+  }
+  return result;
```

Comment:
Why is the added complexity needed?
```suggestion
  if (n === 1) {
    return array;
  }
  return array.filter((element, index) => i % (n + 1) ===0);
```

Reply:
Oh right, the step is _n_...

Reply:
premature optimisation? :D The filter is less efficient.

Reply:
I'm not sure it's premature, since you're calling it from `getEveryNthTick` N times, giving you $O(n^2)$ complexity there. I do wonder whether it's the right optimization though. 😈

Reply:
By switching the step to +n you probably get something like $O(n\cdot\ln n)$ which probably not too bad, but alternatively you could trade that computation and store the tick options.